src/BinSkim.Rules/DwarfRules/BA3003.EnableStackProtector.cs

```diff
@@ -114,20 +126,49 @@ static bool analyze(IDwarfBinary binary, out List<DwarfCompileCommandLineInfo> f
                 return !failedList.Any();
             }
 
+            static bool analyzeSymbols(ElfBinary binary, out List<SymbolEntry<ulong>> failedList)
```

Comment:
>analyzeSymbols [](http://example.com/codeflow?start=24&length=14)

I know you didn't establish this pattern but I am unclear: why are we preferring these static local functions? As opposed to simply using a helper method in this rule?

@shaopeng-gh do you know the history?

Reply:
Not sure about history why we had that style, but the current status may need further improvement, 1. VS will suggest static if possible 2. should add modifier:
![image](https://user-images.githubusercontent.com/81775155/197043764-64240b97-8acf-4cb0-8213-6daeb234489e.png)


What I think normal style should be:
![image](https://user-images.githubusercontent.com/81775155/197042334-751525c4-48b1-48ee-95eb-8a5f97ec1a60.png)